### PR TITLE
[FIX] sale_purchase: access rights error in sale order

### PR DESCRIPTION
**Pull request by alt-odoo**
If a user is only having Sales and not Purchase access,
there is an access right error when computing purchase_line_count
field when viewing the sale order form view because the
compute_sudo is ineffective due to non-stored field.

opw-2056746

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kebeclibre** [comment]
@alt-odoo 
Unless I'm mistaken, the controls to compute can_read are useless, since the user is super user
Also, I think the right fix is to completely hide the button if there are no purchase rights at all

**Comment by alt-odoo** [comment]
@kebeclibre 

The compte_sudo is useless as the computed field is not stored, the user is the current user anyway.

The field purchase_line_count is not a button on a view, and I will have another access rights error if  just restrict it to a specific group.

File: addons/sale_purchase/models/sale_order.py [hunk 1]
```diff
@@ -71,12 +71,14 @@ class SaleOrderLine(models.Model):
     _inherit = 'sale.order.line'
 
     purchase_line_ids = fields.One2many('purchase.order.line', 'sale_line_id', string="Generated Purchase Lines", readonly=True, help="Purchase line generated by this Sales item on order confirmation, or when the quantity was increased.")
-    purchase_line_count = fields.Integer("Number of generated purchase items", compute='_compute_purchase_count', compute_sudo=True)
+    purchase_line_count = fields.Integer("Number of generated purchase items", compute='_compute_purchase_count')
```
Comment (fmdl) [inline addons/sale_purchase/models/sale_order.py]: have you try to add : `groups='purchase.group_purchase_user'` ?

Comment (alt-odoo) [reply addons/sale_purchase/models/sale_order.py]: Yes and we have an "Error: Unknown field purchase_line_count in domain"
due to stock_dropshipping : 
https://github.com/odoo/odoo/commit/6236ed1a5b138c5165929aa3f5c1d1f4f6aa061c#diff-2f6cca7f38255368e4a54b0a19406aa4

**Comment by alt-odoo** [comment]
@simongoffin 
You're right about side effects of having purchase_line_count = 0 in case of no access in your commit : https://github.com/odoo/odoo/commit/6236ed1

We can use sudo and force the company in the read_group instead, see modifications in the commit of this branch.

**Comment by alt-odoo** [comment]
No need to force company as we are using sale line ids, sudo is enough...

**Comment by sswapnesh** [comment]
Will be fixed by https://github.com/odoo/odoo/pull/36247

( Not sure why Two PRs with same fix and OPW reference🙃)

**Pull request closed without merging**